### ISSUE TITLE
Problème de dépendance coop_membership - coop_shift

### ISSUE THREAD

**Issue by remi-filament** [AwesomeFoodCoops/odoo-production#712]
Bonjour,
j'essaie de mettre à jour le code de la Chouette Coop, mais j'ai un problème avec les dépendances entre coop_membership et coop_shift car le premier dépend du deuxième, donc coop_shift sera installé avant.

Le problème vient du fait qu'à plusieurs endroits dans les vues XML (shift_registration_view.xml et menu.xml) il est fait appel aux groupes de sécurité provenant de coop_membership, donc erreur et échec de la mise à jour.

```
2018-10-17 10:08:53,789 1 INFO dbsas openerp.modules.loading: loading coop_shift/views/menu.xml
2018-10-17 10:08:53,819 1 ERROR dbsas openerp.modules.registry: Failed to load registry
Traceback (most recent call last):
  File "/mnt/AwesomeFoodCoops/odoo/openerp/modules/registry.py", line 386, in new
    openerp.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/mnt/AwesomeFoodCoops/odoo/openerp/modules/loading.py", line 374, in load_modules
    force, status, report, loaded_modules, update_module, models_to_check)
  File "/mnt/AwesomeFoodCoops/odoo/openerp/modules/loading.py", line 271, in load_marked_modules
    perform_checks=perform_checks, models_to_check=models_to_check
  File "/mnt/AwesomeFoodCoops/odoo/openerp/modules/loading.py", line 181, in load_module_graph
    _load_data(cr, module_name, idref, mode, kind='data')
  File "/mnt/AwesomeFoodCoops/odoo/openerp/modules/loading.py", line 99, in _load_data
    tools.convert_file(cr, module_name, filename, idref, mode, noupdate, kind, report)
  File "/mnt/AwesomeFoodCoops/odoo/openerp/tools/convert.py", line 849, in convert_file
    convert_xml_import(cr, module, fp, idref, mode, noupdate, report)
  File "/mnt/AwesomeFoodCoops/odoo/openerp/tools/convert.py", line 932, in convert_xml_import
    obj.parse(doc.getroot(), mode=mode)
  File "/mnt/AwesomeFoodCoops/odoo/openerp/tools/convert.py", line 802, in parse
    self._tags[rec.tag](self.cr, rec, de, mode=mode)
  File "/mnt/AwesomeFoodCoops/odoo/openerp/tools/convert.py", line 543, in _tag_menuitem
    group_id = self.id_get(cr, group)
  File "/mnt/AwesomeFoodCoops/odoo/openerp/tools/convert.py", line 780, in id_get
    res = self.model_id_get(cr, id_str, raise_if_not_found)
  File "/mnt/AwesomeFoodCoops/odoo/openerp/tools/convert.py", line 791, in model_id_get
    raise_if_not_found=raise_if_not_found)
  File "/mnt/AwesomeFoodCoops/odoo/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/mnt/AwesomeFoodCoops/odoo/openerp/addons/base/ir/ir_model.py", line 1009, in xmlid_to_res_model_res_id
    return self.xmlid_lookup(cr, uid, xmlid)[1:3]
  File "/mnt/AwesomeFoodCoops/odoo/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "<string>", line 2, in xmlid_lookup
  File "/mnt/AwesomeFoodCoops/odoo/openerp/tools/cache.py", line 85, in lookup
    value = d[key] = self.method(*args, **kwargs)
  File "/mnt/AwesomeFoodCoops/odoo/openerp/addons/base/ir/ir_model.py", line 999, in xmlid_lookup
    raise ValueError('External ID not found in the system: %s' % (xmlid))
ParseError: "External ID not found in the system: coop_membership.group_membership_bdm_lecture" while parsing /mnt/AwesomeFoodCoops/intercoop_addons/coop_shift/views/menu.xml:23, near
<menuitem name="Shifts" parent="shift_main_menu" id="shift_menu" sequence="11" groups="coop_shift.group_shift_user,coop_membership.group_membership_bdm_lecture"/>
2018-10-17 10:08:53,821 1 CRITICAL dbsas openerp.service.server: Failed to initialize database `dbsas`.
Traceback (most recent call last):
  File "/mnt/AwesomeFoodCoops/odoo/openerp/service/server.py", line 894, in preload_registries
    registry = RegistryManager.new(dbname, update_module=update_module)
  File "/mnt/AwesomeFoodCoops/odoo/openerp/modules/registry.py", line 386, in new
    openerp.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/mnt/AwesomeFoodCoops/odoo/openerp/modules/loading.py", line 374, in load_modules
    force, status, report, loaded_modules, update_module, models_to_check)
  File "/mnt/AwesomeFoodCoops/odoo/openerp/modules/loading.py", line 271, in load_marked_modules
    perform_checks=perform_checks, models_to_check=models_to_check
  File "/mnt/AwesomeFoodCoops/odoo/openerp/modules/loading.py", line 181, in load_module_graph
    _load_data(cr, module_name, idref, mode, kind='data')
  File "/mnt/AwesomeFoodCoops/odoo/openerp/modules/loading.py", line 99, in _load_data
    tools.convert_file(cr, module_name, filename, idref, mode, noupdate, kind, report)
  File "/mnt/AwesomeFoodCoops/odoo/openerp/tools/convert.py", line 849, in convert_file
    convert_xml_import(cr, module, fp, idref, mode, noupdate, report)
  File "/mnt/AwesomeFoodCoops/odoo/openerp/tools/convert.py", line 932, in convert_xml_import
    obj.parse(doc.getroot(), mode=mode)
  File "/mnt/AwesomeFoodCoops/odoo/openerp/tools/convert.py", line 802, in parse
    self._tags[rec.tag](self.cr, rec, de, mode=mode)
  File "/mnt/AwesomeFoodCoops/odoo/openerp/tools/convert.py", line 543, in _tag_menuitem
    group_id = self.id_get(cr, group)
  File "/mnt/AwesomeFoodCoops/odoo/openerp/tools/convert.py", line 780, in id_get
    res = self.model_id_get(cr, id_str, raise_if_not_found)
  File "/mnt/AwesomeFoodCoops/odoo/openerp/tools/convert.py", line 791, in model_id_get
    raise_if_not_found=raise_if_not_found)
  File "/mnt/AwesomeFoodCoops/odoo/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/mnt/AwesomeFoodCoops/odoo/openerp/addons/base/ir/ir_model.py", line 1009, in xmlid_to_res_model_res_id
    return self.xmlid_lookup(cr, uid, xmlid)[1:3]
  File "/mnt/AwesomeFoodCoops/odoo/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "<string>", line 2, in xmlid_lookup
  File "/mnt/AwesomeFoodCoops/odoo/openerp/tools/cache.py", line 85, in lookup
    value = d[key] = self.method(*args, **kwargs)
  File "/mnt/AwesomeFoodCoops/odoo/openerp/addons/base/ir/ir_model.py", line 999, in xmlid_lookup
    raise ValueError('External ID not found in the system: %s' % (xmlid))
ParseError: "External ID not found in the system: coop_membership.group_membership_bdm_lecture" while parsing /mnt/AwesomeFoodCoops/intercoop_addons/coop_shift/views/menu.xml:23, near
<menuitem name="Shifts" parent="shift_main_menu" id="shift_menu" sequence="11" groups="coop_shift.group_shift_user,coop_membership.group_membership_bdm_lecture"/>
```

Pour contourner le problème, je vais supprimer les dépendances à coop_membership dans coop_shift (donc supprimer les droits des menus) le temps de l'installation, mais tous ceux qui voudront installer le module coop_membership pour la première fois devraient avoir ce problème.

**Comment by simonjejcic** [comment]
Hello

Merci du message, on va regarder ça

Simon

Le mer. 17 oct. 2018 à 12:37, remi-filament <notifications@github.com> a
écrit :

**Comment by real34** [comment]
Bonjour,

Est-ce qu’il est possible d’aider à notre échelle sur ce problème ? Le fait qu’il n’y ait pas eu d’autres intéractions sur cette demande est-il lié au fait que le besoin soit suivi dans un autre système ?

Bonne journée.

**Comment by simonjejcic** [comment]
Hello

Oui nous avons un ticket mais pas de date de résolution.
Vous pouvez faire une PR si vous le souhaitez.

@+

Le dim. 11 nov. 2018 09:53, Pierre Martin <notifications@github.com> a
écrit :

**Comment by simonjejcic** [comment]
Hello,

I see ticket solved end October.
Would you please continue in English in case not
Thx.

**Comment by real34** [comment]
Hi Simon,

Thank you for this quick feedback.
Do you think it could be bf36d09c445b976a88ea32428d62d485f9a936ab (PR https://github.com/AwesomeFoodCoops/odoo-production/pull/732) that fixes the issue?

If yes we will try it with the latest branch and close the issue.

Thanks.

**Comment by simonjejcic** [comment]
Yes

Le dim. 11 nov. 2018 11:17, Pierre Martin <notifications@github.com> a
écrit :

**Comment by metaminux** [comment]
Hi Simon,

I'm working for Otsokop.
Sorry to reopen this subject one year later, but there are still inter-dependencies between coop_shift and coop_membership.

2 files in coop_shift are using groups from coop_membership :
* https://github.com/AwesomeFoodCoops/odoo-production/blob/8c8e878a784f3b84b9d25aaf728179b4d7b6d19b/intercoop_addons/coop_shift/views/shift_registration_view.xml#L35
* https://github.com/AwesomeFoodCoops/odoo-production/blob/8c8e878a784f3b84b9d25aaf728179b4d7b6d19b/intercoop_addons/coop_shift/views/shift_registration_view.xml#L50-L52
* https://github.com/AwesomeFoodCoops/odoo-production/blob/8c8e878a784f3b84b9d25aaf728179b4d7b6d19b/intercoop_addons/coop_shift/views/shift_registration_view.xml#L59
* https://github.com/AwesomeFoodCoops/odoo-production/blob/8c8e878a784f3b84b9d25aaf728179b4d7b6d19b/intercoop_addons/coop_shift/security/ir.model.access.csv#L65

Will maybe try to create a new PR

**Comment by simonjejcic** [comment]
Hello,

Please don't create a PR, this code is already migrated, we'll check in
version 12

Thanks!

Le mar. 5 nov. 2019 à 02:49, Guillaume Masson <notifications@github.com> a
écrit :

**Comment by metaminux** [comment]
Hi Simon,

Thanks for the answer, but which version 12 are you talking about ?
Are you migrating to Odoo 12.0 ? I can't see anything about that on GH...

**Comment by zocario** [comment]
Hi @simonjejcic / @msdruidoo / @ivantodorovich / @lehoangan, how can we fix this issue? We need to install those modules in production. Can you clarify what you meant by `version 12`? We are using Odoo 9 at Otsokop so we are not sure about the direction to take..

**Comment by paulRbr** [comment]
@zocario @metaminux I believe Simon is talking about the work they are doing on the [“duidoo" fork](https://github.com/druidoo/FoodCoops/tree/12.0) which has a `12.0` branch.

However if most of the foodcoops using this codebase are still on Odoo 9.0 version I think it would be good for most of them to have a fix on the inter-dependencies between `coop_shift` and `coop_membership` modules land on the `9.0` branch.

I got bitten by the exact same problem while trying to install this project 😅

**Comment by simonjejcic** [comment]
Hi Paul!
12.0 is now the version to use for new projects. It's still beta, it will
be pushed on awesomefoodcoops very soon ;)
Take care

**Comment by paulRbr** [comment]
> Hi Paul! 12.0 is now the version to use for new projects. It's still beta, it will be pushed on awesomefoodcoops very soon ;) Take care Le jeu. 30 avr. 2020 à 01:05, Paul B. 

It's a good thing that you folks have been working on a compatibility with Odoo12.0, however if this is still a work in progress (“still beta” as you said), then it would be good to fix the “stable” (current 9.0) version of this project :slightly_smiling_face: so that people trying it out don't get bitten by this bug.

I have been committing some of my discoveries, so https://github.com/paulRbr/FoodCoops/commit/17eb0209f06c3a86affc003dd196d23a9e269070 will help anybody searching to run the current default branch `9.0` of this repo. (I don't know what to do next with this change though, if I should revert the commit once the Odoo instance was installed with all the modules for instance?)

**Comment by remi-filament** [comment]
The problem is still present and actually the views may not be a problem, the main issue is related to the last line in ir.model.access.csv which adds an access rule for a coop_shift module based on coop_membership group.
Easy fix is to remove these lines which should go to coop_membership module instead
